frontend/src/features/apartment/ApartmentSalesPage.tsx

```diff
@@ -0,0 +1,296 @@
+import MaximumPriceModalContent from "./components/ApartmentMaximumPriceBreakdownModal";
+
+const MaximumPriceModalError = ({error, setIsModalVisible}) => {
+    const nonFieldError = ((error as FetchBaseQueryError)?.data as {message?: string})?.message || "";
+    return (
+        <>
+            <Dialog.Content>
+                <p>Virhe: {(error as FetchBaseQueryError)?.status}</p>
+                <p>{nonFieldError}</p>
+            </Dialog.Content>
+            <Dialog.ActionButtons>
+                <Button
+                    onClick={() => setIsModalVisible(false)}
+                    variant="secondary"
+                    theme="black"
+                >
+                    Sulje
+                </Button>
+            </Dialog.ActionButtons>
+        </>
+    );
+};
+
+const LoadedApartmentSalesPage = ({apartment}: {apartment: IApartmentDetails}) => {
+    const [isModalVisible, setIsModalVisible] = useState(false);
+    const hasValidCalculation = apartment.prices.maximum_prices.confirmed?.valid.is_valid;
+    const initialFormData = {
+        announcementDate: null,
+        dealDate:
+            apartment.prices.maximum_prices.confirmed !== null
+                ? apartment.prices.maximum_prices.confirmed.calculation_date
+                : null,
+        dateOfCount:
+            apartment.prices.maximum_prices.confirmed !== null
+                ? apartment.prices.maximum_prices.confirmed.calculation_date
+                : null,
+        price: null,
+        apartment_share_of_housing_company_loans: hasValidCalculation ? apartment.prices.primary_loan_amount : null,
+        mortgageShare: null,
+        isExcludedFromStatistics: false,
+        index: "",
+    };
+    const [isExcludedFromStatistics, setIsExcludedFromStatistics] = useState(false);
+    const [formData, setFormData] = useImmer(initialFormData);
+    const [formOwnershipsList, setFormOwnershipsList] = useImmer<(IOwnership & {key: string})[]>(
+        apartment !== undefined ? apartment.ownerships.map((o) => ({...o, key: uuidv4()})) : []
+    );
+    const [saveMaximumPrice, {data, error, isLoading}] = useSaveApartmentMaximumPriceMutation();
+    const {data: maxPriceQuery, error: maxPriceError} = useGetApartmentMaximumPriceQuery({
+        housingCompanyId: apartment.links.housing_company.id,
+        apartmentId: apartment.id,
+        priceId: apartment.prices.maximum_prices.confirmed?.id as string,
+    });
+    const [maxPrices, setMaxPrices] = useState({
+        maxPrice: apartment.prices.maximum_prices.confirmed?.maximum_price,
+        maxPricePerSquare: apartment.prices.maximum_prices.confirmed?.maximum_price
+            ? apartment.prices.maximum_prices.confirmed?.maximum_price / apartment.surface_area
+            : 0,
+        debtFreeMaxPrice: apartment.prices.debt_free_purchase_price,
+        loanAmount: apartment.prices.primary_loan_amount,
+        index: data ? data.index : maxPriceQuery?.index,
+    });
```

Comment:
Same as above:

**_(suggested names are for JS naming scheme, but to simplify the process of sending data to the API, you can use the same variable names in formData as we have done in most other forms)_**

`maxPrice` => `maximumPrice`
`maxPricePerSquare` We should refrain from doing these calculations on the frontend, but this one is not yet returned from the API, created a ticket for this (HT-411)
`debtFreeMaxPrice` => `debtFreePurchasePrice`
`loanAmount` => `primaryLoanAmount`